### PR TITLE
Личный проект: пока все дома

### PR DESCRIPTION
**Pull request by rule-z**
---
:mortar_board: [Личный проект: пока все дома](https://up.htmlacademy.ru/javascript/15/user/34650/tasks/13)

:boom: https://htmlacademy-javascript.github.io/34650-keksobooking/2/

### CODE REVIEW

File: js/map.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+var getRandomOfferCheckinsNumber = function () {
+  return getRandomNumber(0, OFFER_CHECKINS.length - 1);
+};
+
+var getRandomOfferCheckinsTime = function () {
+  return OFFER_CHECKINS[getRandomOfferCheckinsNumber()];
+};
+
+var getRandomOfferCheckoutsNumber = function () {
+  return getRandomNumber(0, OFFER_CHECKOUTS.length - 1);
```
Comment (rkhaslarov) [inline js/map.js]: As I mentioned before, don't wrap function with args, simply use it directly.

File: js/map.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+};
+
+var getRandomOfferCheckoutsTime = function () {
+  return OFFER_CHECKOUTS[getRandomOfferCheckoutsNumber()];
+};
+
+var getShuffledArray = function (arrayName) {
```
Comment (rkhaslarov) [inline js/map.js]: arrayName? rename it to array

File: js/map.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  mapCardElement.querySelector('.popup__description').textContent = pin.offer.description;
+  mapCardElement.querySelector('.popup__photos').content.querySelector('.map__photo').src = pin.offer.photos;
+
+  return mapCardElement;
+};
+
+var createMapCardNodes = function () {
+  var fragment = document.createDocumentFragment();
+
+  for (var i = 0; i < 8; i++) {
```
Comment (rkhaslarov) [inline js/map.js]: 8 magic number

File: js/map.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+var renderMapCard = function (pin) {
+  var mapCardElement = CARD_ITEM.cloneNode(true);
+
+  mapCardElement.querySelector('.popup__avatar').src = pin.author.avatar;
+  mapCardElement.querySelector('.popup__title').textContent = pin.offer.title;
+  mapCardElement.querySelector('.popup__text--address').textContent = pin.offer.address;
+  mapCardElement.querySelector('.popup__text--price').textContent = pin.offer.price;
+  mapCardElement.querySelector('.popup__type').textContent = pin.offer.type;
+  mapCardElement.querySelector('.popup__text--capacity').textContent = pin.offer.rooms + ' комнаты для ' + pin.offer.guests + ' гсотей';
```
Comment (rkhaslarov) [inline js/map.js]: гсотей ?

File: js/map.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+  var fragment = document.createDocumentFragment();
+
+  for (var i = 0; i < 8; i++) {
+    fragment.appendChild(renderMapCard(generateMapPinInfo()));
+  }
+
+  return fragment;
+};
+
+CARD_ITEM.appendChild(MAP_WINDOW.insertBefore(createMapCardNodes(), FILTERS_CONTAINER));
```
Comment (rkhaslarov) [inline js/map.js]: Try something like this
mapPins.appendChild(renderArrayToChildNodes(advertisementsArray, renderPin));
mapSection.insertBefore(renderCard(advertisementsArray[0]), document.querySelector('.map__filters-container'));

Comment (rkhaslarov) [reply js/map.js]: Just as example

File: js/map.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+  mapCardElement.querySelector('.popup__avatar').src = pin.author.avatar;
+  mapCardElement.querySelector('.popup__title').textContent = pin.offer.title;
+  mapCardElement.querySelector('.popup__text--address').textContent = pin.offer.address;
+  mapCardElement.querySelector('.popup__text--price').textContent = pin.offer.price;
+  mapCardElement.querySelector('.popup__type').textContent = pin.offer.type;
+  mapCardElement.querySelector('.popup__text--capacity').textContent = pin.offer.rooms + ' комнаты для ' + pin.offer.guests + ' гсотей';
+  mapCardElement.querySelector('.popup__text--time').textContent = 'Заезд после ' + pin.offer.checkin + ', выезд до ' + pin.offer.checkout;
+  mapCardElement.querySelector('.popup__features').textContent = pin.offer.features;
+  mapCardElement.querySelector('.popup__description').textContent = pin.offer.description;
+  mapCardElement.querySelector('.popup__photos').content.querySelector('.map__photo').src = pin.offer.photos;
```
Comment (rkhaslarov) [inline js/map.js]: For features and photos try next, it's a pseudo code not only 
for (var i = 0; i < features.length; i++) {
    featuresContainer.innerHTML += '<li class="feature feature--' + features[i] + '"></li>';
  }

File: js/map.js (hunk 2 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 4 above)
Comment (rule-z) [reply js/map.js]: Is done! 🤣

File: js/map.js (hunk 6 above)
Comment (rule-z) [reply js/map.js]: Photos is done! Features last

File: js/map.js (hunk 1 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 3 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js [hunk 7]
```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+var mapSection = document.querySelector('.map');
+mapSection.classList.remove('map--faded');
+
+var mapPins = document.querySelector('.map__pins');
+var pinItem = document.querySelector('.map__pin');
```
Comment (rkhaslarov) [inline js/map.js]: Here you should get template var pinItem = document.querySelector('#pin').content; not particular pin

File: js/map.js [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+};
+
+var renderMapCard = function (pin) {
+  var mapCardElement = cardItem.cloneNode(true);
+
+  mapCardElement.querySelector('.popup__avatar').src = pin.author.avatar;
+  mapCardElement.querySelector('.popup__title').textContent = pin.offer.title;
+  mapCardElement.querySelector('.popup__text--address').textContent = pin.offer.address;
+  mapCardElement.querySelector('.popup__text--price').textContent = pin.offer.price + '₽';
+  // mapCardElement.querySelector('.popup__type').textContent = pin.offer.type;
```
Comment (rkhaslarov) [inline js/map.js]: Remove this part

File: js/map.js [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+    if (typeEn === OFFER_TYPES[i]) {
+      var result = i;
+    }
+  }
+
+  return result;
+};
+
+var getTranslatedType = function (type) {
+  return OFFER_TYPES_RU[getEnTypeIndex(type)];
```
Comment (rkhaslarov) [inline js/map.js]: use object instead of additional array

File: js/map.js [hunk 10]
```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+var mapSection = document.querySelector('.map');
+mapSection.classList.remove('map--faded');
+
+var mapPins = document.querySelector('.map__pins');
+var pinItem = document.querySelector('.map__pin');
+var mapFiltersContainer = document.querySelector('.map__filters-container');
+var cardItem = document.querySelector('#card').content.querySelector('.map__card');
+var cardsAndPinsCount = 8;
```
Comment (rkhaslarov) [inline js/map.js]: use constant naming convention

File: js/map.js [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+var OFFER_TYPES = ['palace', 'flat', 'house ', 'bungalo'];
+var OFFER_TYPES_RU = ['Дворец', 'Квартира', 'Дом', 'Бунгало'];
+var OFFER_CHECKINS = ['12:00', '13:00', '14:00'];
+var OFFER_CHECKOUTS = ['12:00', '13:00', '14:00'];
+var OFFER_FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+var OFFER_PHOTOS = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+];
```
Comment (rkhaslarov) [inline js/map.js]: Move all constants to the beginning of file

File: js/map.js [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+
+  for (var i = 0; i < cardsAndPinsCount; i++) {
+    fragment.appendChild(renderMapCard(generateMapPinInfo()));
+  }
+
+  return fragment;
+};
+
+
+// CARD_ITEM.appendChild(MAP_WINDOW.insertBefore(createMapCardNodes(), FILTERS_CONTAINER));
```
[the remark this answers is not in the document]
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 7 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 8 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 10 above)
Comment (rule-z) [reply js/map.js]: Is done!

File: js/map.js (hunk 11 above)
Comment (rule-z) [reply js/map.js]: Is done!

**Pull request merged**